christfellowshipchurch/apollos-app
# [Chat] Visual Cleanup - Round 1

### Summary

This PR fixes the following random visual issues in Chat.

1. Chat left/right padding inconsistent ([Basecamp](https://3.basecamp.com/3926363/buckets/17566548/todos/3062988841])).

1. Update the list of reaction emojis ([Basecamp](https://3.basecamp.com/3926363/buckets/17566548/todos/3061355416)).

1. Shrink message text input ([Basecamp](https://3.basecamp.com/3926363/buckets/17566548/todos/3061713067)) and enlarge Send button ([Basecamp](https://3.basecamp.com/3926363/buckets/17566548/todos/3061708579)).

1. Update background for ReactionsList and ReactionPicker ([Basecamp](https://3.basecamp.com/3926363/buckets/17566548/todos/3061415668)).

1. Change how to take actions on messages and users ([Basecamp](https://3.basecamp.com/3926363/buckets/17566548/todos/3061354110)).
    - Tapping on a message brings up ReactionPicker.
    - Long pressing on a message brings up action sheet with message-only actions.
    - Long pressing on an avatar brings up action sheet with user-only actions.

1. Make it easier to dismiss keyboard ([Basecamp](https://3.basecamp.com/3926363/buckets/17566548/todos/3061352504)).


### Discussion

1. ~Using an `envelope` icon for the Send button temporarily. When we get a `paper-airplane` icon, will swap it in in a future PR.~

### Testing

(Same items as above.)

1. Check that left/right padding in main messages window is even.
1. Check that the reactions emoji list is what is in the Basecamp todo (only positive emojis).
1. Thoughts on the new message input?
1. Check that the background color for the ReactionsList and ReactionPicker is light in light mode and dark in dark mode.
1. Check that tapping a message brings up reaction picker. Check that long pressing messages bring up message actions and long pressing avatars brings up user actions. Note that long pressing on yourself does nothing because there are no user actions to take on yourself.
1. Play around with dismissing the keyboard to see how it feels.


### Screenshots

Correct padding, new emojis, new text input, correct reactions background.

|Light mode|Dark mode|
|---|---|
|<img width="434" alt="Screen Shot 2020-09-28 at 11 07 40 PM" src="https://user-images.githubusercontent.com/200488/94583626-ab9e1980-024b-11eb-9521-c5a58543f3f4.png">|<img width="434" alt="Screen Shot 2020-09-29 at 9 42 28 AM" src="https://user-images.githubusercontent.com/200488/94583641-af31a080-024b-11eb-8f62-a477d728d4a3.png">|
